stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -1309,7 +1311,80 @@ public static void populateSenderConfigurationOptions(SenderConfiguration sender
         }
     }
 
-    private static boolean isInteger(long val) {
+    public static void populateSenderConfigurations(SenderConfiguration senderConfiguration, Struct
+            clientEndpointConfig) {
+        ProxyServerConfiguration proxyServerConfiguration;
+        Struct secureSocket = clientEndpointConfig.getStructField(HttpConstants.ENDPOINT_CONFIG_SECURE_SOCKET);
+
+        if (secureSocket != null) {
+            HttpUtil.populateSSLConfiguration(senderConfiguration, secureSocket);
+        } else {
+            HttpUtil.setDefaultTrustStore(senderConfiguration);
+        }
+        Struct proxy = clientEndpointConfig.getStructField(HttpConstants.PROXY_STRUCT_REFERENCE);
+        if (proxy != null) {
+            String proxyHost = proxy.getStringField(HttpConstants.PROXY_HOST);
+            int proxyPort = (int) proxy.getIntField(HttpConstants.PROXY_PORT);
+            String proxyUserName = proxy.getStringField(HttpConstants.PROXY_USERNAME);
+            String proxyPassword = proxy.getStringField(HttpConstants.PROXY_PASSWORD);
+            try {
+                proxyServerConfiguration = new ProxyServerConfiguration(proxyHost, proxyPort);
+            } catch (UnknownHostException e) {
+                throw new BallerinaConnectorException("Failed to resolve host" + proxyHost, e);
+            }
+            if (!proxyUserName.isEmpty()) {
+                proxyServerConfiguration.setProxyUsername(proxyUserName);
+            }
+            if (!proxyPassword.isEmpty()) {
+                proxyServerConfiguration.setProxyPassword(proxyPassword);
+            }
+            senderConfiguration.setProxyServerConfiguration(proxyServerConfiguration);
+        }
+
+        String chunking = clientEndpointConfig.getRefField(HttpConstants.CLIENT_EP_CHUNKING).getStringValue();
+        senderConfiguration.setChunkingConfig(HttpUtil.getChunkConfig(chunking));
+
+        long timeoutMillis = clientEndpointConfig.getIntField(HttpConstants.CLIENT_EP_ENDPOINT_TIMEOUT);
+        if (timeoutMillis < 0 || !isInteger(timeoutMillis)) {
+            throw new BallerinaConnectorException("invalid idle timeout: " + timeoutMillis);
```

Comment:
Instead of throwing a connector exception wouldn't it be better to use a default value like int_max because long is the int in Ballerina. We can log a warning though as done here: https://github.com/riyafa/ballerina/blob/d0d7802091db6e5e699349d29e863a04085d471c/stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java#L221-L231